test/apivalidations/flavor_test.go

```diff
@@ -113,7 +113,7 @@ var _ = Describe("ORC Flavor API validations", func() {
 		patch := baseFlavorPatch(flavor)
 		maxString := strings.Repeat("a", 65536)
 		patch.Spec.WithResource(applyconfigv1alpha1.FlavorResourceSpec().WithRAM(1).WithVcpus(1).WithDescription(maxString))
-		Expect(applyObj(ctx, flavor, patch)).NotTo(Succeed())
+		Expect(applyObj(ctx, flavor, patch)).To(MatchError(ContainSubstring("spec.resource.description: Too long")))
```

Comment:
I won't block it, but I have never liked testing the contents of error strings. I have never found value in it, but have frequently found them to be a source of churn and noise. We're now tightly coupled to the exact format of error strings emitted by kube-apiserver. This isn't a formal API, and they will break it. It will unnecessarily constrain the versions of kubernetes our tests can be executed against.

If there are any formal semantics in the error message I'd be happy to test those, but I don't like just testing strings intended for human consumption.

Reply:
There was a reason for using matcher that checked for the error: when I made `disk` a required field, many of the test remained green although the validations failed for the wrong reasons. We were not testing what we thought we were testing, which is a much greater risk than our tests starting to fail because k8s changed how they report errors. I think it's a good tradeoff.